include/beast/websocket/detail/decorator.hpp

```diff
@@ -40,6 +41,32 @@ class decorator : public abstract_decorator
 {
     T t_;
 
+    class call_req_possible
+    {
+        template<class U, class R = decltype(
+            std::declval<U>().operator()(
+                std::declval<request_type&>()),
+                    std::true_type{})>
+        static R check(int);
+        template<class>
+        static std::false_type check(...);
+    public:
+        using type = decltype(check<T>(0));
+    };
+
+    class call_res_possible
+    {
+        template<class U, class R = decltype(
+            std::declval<U>().operator()(
+                std::declval<response_type&>()),
+                    std::true_type{})>
+        static R check(int);
+        template<class>
+        static std::false_type check(...);
+    public:
+        using type = decltype(check<T>(0));
+    };
+
```

Comment:
@HowardHinnant If I change it then I would have to change all the others which don't use `is_call_possible`...